<launch>

  <!-- Sets the image source. -->
  <arg name="image_topic" default="/camera/rgb/image_raw" />
  
  
  <!-- Defines which and where are the markers in the environment. The sample
    file (markers_configuration_sample.yml) provides complete documentation.
    -->
  <arg name="markers_configuration" 
    default="$(find ros_markers)/config/markers_configuration_sample.yml" />
  
  <!-- Sets whether or not to ignore markers not included in the 
	configuration file.    -->
  <arg name="omit_other_tags" default="false" />	
	
  <!-- Sets the default size of markers, in millimeters, which are not 
    in the configuration file. -->
  <arg name="default_marker_size" default="28.5" />
    	
  <!-- set the level of filtering applied on the detected markers. 1.0 means
    no filtering, while 0.0 is the maximum level of filtering. 
    (-1 uses chilitags default value). -->
  <arg name="gain" default="-1" />

  <!-- set the number of frames an object/marker should be not detected
    for before it is removed. (-1 uses chilitags default value). -->
  <arg name="persistence" default="-1" />
  
  <node pkg="ros_markers" type="detect" name="ros_markers">
    <remap from="image" to="$(arg image_topic)" />
    <param name="markers_configuration" type="str" 
      value="$(arg markers_configuration)" />
    <param name="omit_other_tags" type="bool" value="$(arg omit_other_tags)" />
    <param name="default_marker_size" type="double" value="$(arg default_marker_size)" />
    <param name="gain" type="double" value="$(arg gain)" />
    <param name="persistence" type="int" value="$(arg gain)" />
  </node>

</launch>
